### DISCUSSION TITLE
Not working with PyCharm: "An error occurred while fetching the completions. Please try again."

### DISCUSSION THREAD

**Discussion by gsusI** [community/community#14537] [Copilot Conversations]
Github Copilot doesn't generate any suggestions in PyCharm, no matter in which file/line my cursor is, it always shows the same error:

![image](https://user-images.githubusercontent.com/4558016/162393366-631fe173-ba6c-4a2d-beed-914f2618fddd.png)

My list of plugins: 
```txt
Cloud Code (22.3.1-213)
GitHub Copilot (1.1.15.1147-agent)
Node.js (213.6461.6)
PowerShell (2.0.10)
Requirements (2022.3-2021.3)
Tabnine AI Code Completion- JS Java Python TS Rust Go PHP & More (0.5.23)
Terraform and HCL (0.7.15)
VSCode Keymap (213.5744.3)
Database Tools and SQL (213.7172.26)
Docker (213.7172.26)
FTP/SFTP Connectivity (ex. Remote Hosts Access) (213.7172.26)
Puppet (213.7172.26)
Vagrant (213.7172.26)
HTML Tools (213.7172.26)
IDE Settings Sync (213.7172.26)
Settings Repository (213.7172.26)
Angular and AngularJS (213.7172.26)
CoffeeScript (213.7172.26)
JavaScript and TypeScript (213.7172.26)
JavaScript Debugger (213.7172.26)
JavaScript Intention Power Pack (213.7172.26)
TSLint (213.7172.26)
Gherkin (213.7172.26)
GNU GetText files support (*.po) (213.7172.26)
Ini (213.7172.26)
Markdown (213.7172.26)
Properties (213.7172.26)
Shell Script (213.7172.26)
YAML (213.7172.26)
Endpoints (213.7172.26)
Remote Development Gateway (213.7172.26)
CSS (213.7172.26)
Less (213.7172.26)
Sass (213.7172.26)
Stylus (213.7172.26)
Haml (213.7172.26)
Windows 10 Light Theme (213.7172.26)
ChangeReminder (213.7172.26)
Git (213.7172.26)
GitHub (213.7172.26)
Mercurial (213.7172.26)
Perforce Helix Core (213.7172.26)
Subversion (213.7172.26)
Code With Me (213.7172.26)
Copyright (213.7172.26)
Diagrams (213.7172.26)
EditorConfig (213.7172.26)
File Watchers (213.7172.26)
Grazie (213.7172.26)
HTTP Client (213.7172.26)
IDE Features Trainer (213.7172.26)
IDE Features Trainer: Git Lessons (213.7172.26)
IntelliLang (213.7172.26)
Machine Learning Code Completion (213.7172.26)
Machine Learning in Search Everywhere (213.7172.26)
Performance Testing (213.7172.26)
ReStructuredText (213.7172.26)
Shared Project Indexes (213.7172.26)
Space (213.7172.26)
Task Management (213.7172.26)
Terminal (213.7172.26)
TextMate Bundles (213.7172.26)
Time Tracking (213.7172.26)
Toml (213.7172.26)
```

**Comment by jansorg** [comment]
Are you on Windows? The -agent nightly builds are currently broken on Windows, but will be fixed soon.

**Reply by klevamane** [reply]
I am currently facing the same issue right now. I'm making use of Intel Macbook pro

**Reply by klevamane** [reply]
@gsusI did you get a solution to this?

**Reply by klevamane** [reply]
Oh never mind, i needed to update it. It was showing the errors on the Event log

**Reply by SMZ70** [reply]
thanks @klevamane. update solved my issue

**Comment by affirmatives** [comment]
me too , idea 2021.2 its not work

**Reply by Raxcl** [reply]
+1

**Reply by serdnago** [reply]
+2

**Reply by yassansplus** [reply]
+3

**Reply by giuliotosato** [reply]
+4

**Reply by CaptainWuDaoKou** [reply]
+5

**Reply by yxqweasd** [reply]
try upgrade the copilot extension version.

**Comment by YeStone1** [comment]
1ALX2jkRUVUbQE8RsLx5j6VkrEJMrhihin



чт, 23 июн. 2022 г., 17:44 Huang Weinan ***@***.***>:

**Comment by fuwiak** [comment]
After updating Pycharm and copilot I have the same problem. I installed copilot from disk. Despite repeated reboots, it still does not work. Its no other option?

**Comment by zlingbo** [comment]
It's not working  on PyCharm 2022.2.1 (Professional Edition) ,too.

**Comment by alansastre** [comment]
Same problem both in PyCharm Professional and DataSpell in ipynb files.

**Comment by CapsuleCatt** [comment]
Uhhhhh I got the same problem as well... PyCharm 2022.2.1 and the latest version of copilot...

**Comment by Zemelware** [comment]
I have the same issue on PyCharm 2021.3.3 Professional

**Comment by solsab** [comment]
Same error. It doesn't work.

**Comment by TankNee** [comment]
I have the same issue on PyCharm 2022.2.3 Professional Edition.
Platform: MacBook Pro 14 M1 Pro 8+14
OS: macOS 12.6

**Comment by sajjadali54** [comment]
Hi @gsusI 

GitHub Copilot is a tool that uses machine learning to suggest code completions and documentation snippets based on your current context. If it is not generating any suggestions in PyCharm, there could be a few possible reasons for this:

1. There may be a problem with the GitHub Copilot configuration in PyCharm. You can try re-enabling the plugin or checking that it is properly configured.

2. The GitHub Copilot model may not be trained on the type of code you are working on. GitHub Copilot uses a machine learning model that is trained on a large dataset of open-source code, so it may not be able to provide suggestions for code that is significantly different from the training data.

3. There may be connectivity issues that are preventing the plugin from communicating with the GitHub API. Make sure that you have an active internet connection and that you have authorized the plugin to access your GitHub account.

If none of these suggestions resolve the issue, you may want to try reaching out to the GitHub Copilot team or checking their documentation for additional troubleshooting tips.

I hope this information is helpful! If you have any additional questions, please don't hesitate to ask.

**Comment by petersodhiyahoocouk** [comment]
Ok I had the same issue. I just went back to the main code and waited and co-pilot was enabled.

**Comment by Weinwurm** [comment]
This happens, when your code has forbidden words in it. 
I had to work with  medical data containing the so incredible offensive word "gender." And Copilot stopped working till I had renamed it (now we talk about "bender").
Nobody knows which words are blacklisted, so you can just find out by trial and error. 
What a time to be alive!

**Comment by DanielAvdar** [comment]
![cp-error](https://github.com/user-attachments/assets/3f27c461-f5f1-4827-bd9f-f29cb0433a7f)
PyCharm Copilot: Frequent errors, works briefly after restart (10 min), issue persistent for 2-3 weeks.

**Reply by BlueSkyLT** [reply]
Same issue here. Tried unintalling, restarting, and updating. None has work. Persistent for about 2 weeks.

**Comment by DanielAvdar** [comment]
Failed to initiate the GitHub login process. Please try again.
